src/sage/combinat/constellation.py

```diff
@@ -1047,6 +1043,37 @@ def __iter__(self):
                 continue
             yield self(list(p) + [None], check=False)
 
+    def cardinality(self):
+        r"""
+        Return the cardinality of ``self``.
+
+        EXAMPLES:
+
+        One can check the first few terms of sequence :oeis:`A220754`::
+
+            sage: [Constellations(4, k).cardinality() for k in range(1, 6)]
+            [1, 7, 194, 12858, 1647384]
+
+            sage: [Constellations(3, k, connected=False).cardinality() for k in range(1, 6)]
+            [1, 4, 36, 576, 14400]
+        """
+        k = self._length
+        if not k:
+            return ZZ.one()
+
+        if not self._connected:
+            return factorial(self._degree) ** (k-1)
+
+        @cached_function
```

Comment:
I suspect the caching is only local, since each time `a` is created anew.

If you intend to throw away the cache on each call, might as well use an array and populate it in a loop (also avoid possible recursion error). If not, maybe you can move `a` to a global private function.